iup/eval_objects.py

```diff
@@ -0,0 +1,288 @@
+import abc
+import polars as pl
+from abc import abstractmethod
+
+#### prediction output ####
+class Forecast(pl.DataFrame, metaclass=abc.ABCMeta):
+    """
+    Abstract class for all the forecast data type.
+    """
+    def __init__(self, *args, **kwargs):
+
+        super().__init__(*args, **kwargs)
+        self.validate()
+
+    @abc.abstractmethod
+    def validate(self) -> None:
+        pass
+
+
+class PointForecast(Forecast):
+    """
+    Class for forecast with point estimate
+    Can be a special case for QuantileForecast.
+
+    """
+
+    def __init__(self):
```

Comment:
When you define `validate()` in `PointForecast`, it overrides the definition of `validate()` that was inherited from `Forecast`. So `super().__init__()` calls the same `.validate()` as the line below this does

Prove it to yourself! See this example: https://gist.github.com/swo/4c641bdfbd9d805d42542d7c7290fd0b

Reply:
Because I have `self.validate()` followed by `super().__init__()` for each class, this means it is the `validate()` in this class gets called instead of its direct parent class. This will result in that the `validate()` in the ancestor class (here is `ValidateData`) never gets called, but it is fine because it is an abstract class and doesn't have defined `validate()`. Is that correct? The correct way is to delete `self.validate()` for each class.